skabase/SKALogger/SKALogger/SKALogger.py

```diff
@@ -128,33 +130,86 @@ def Log(self, argin):
+            self.info_stream("Element Logging level : %s, Device : %s", ElementLoggingLevel[i], ElementLoggingDevice[i])
+            logger.debug("Element Logging level : %s, Device : %s", ElementLoggingLevel[i], ElementLoggingDevice[i])
+            logger.info("Element Logging level : %s, Device : %s", ElementLoggingLevel[i], ElementLoggingDevice[i])
+            logger.warning("Element Logging level : %s, Device : %s", ElementLoggingLevel[i], ElementLoggingDevice[i])
+            logger.error("Element Logging level : %s, Device : %s", ElementLoggingLevel[i], ElementLoggingDevice[i])
+            logger.fatal("Element Logging level : %s, Device : %s", ElementLoggingLevel[i], ElementLoggingDevice[i])
+            dev1 = DeviceProxy(ElementLoggingDevice[i])
+            dev1.elementLoggingLevel = ElementLoggingLevel[i]
+            property_names = ["logging_level",
+                              "logging_target",
+                              "ElementLogger",
+                              "ElementLoglevel",
+                              "ServerHostName",
+                              ]
+            dev_properties = dev1.get_property(property_names)
+            dev_properties["logging_level"] = ["INFO"]
+            dev_properties["logging_target"] = ["device::localhost:10123/ref/elt/logger"]
+            dev_properties["ElementLogger"] = ["device::localhost:10123/ref/elt/logger"]
+            dev_properties["ElementLoglevel"] = [ElementLoggingLevel[i]]
+            dev_properties["ServerHostName"] = ["localhost"]
+            dev1.put_property(dev_properties)
+            dev1.add_logging_target("device::localhost:10123/ref/elt/logger")
+            i += 1
         # PROTECTED REGION END #    //  SKALogger.SetElementLoggingLevel
 
     @command(
-    dtype_in='str', 
+    dtype_in='DevVarLongStringArray', 
     doc_in="Storage logging level for selected devices", 
     )
     @DebugIt()
     def SetStorageLoggingLevel(self, argin):
         # PROTECTED REGION ID(SKALogger.SetStorageLoggingLevel) ENABLED START #
-        pass
+        StorageLoggingLevel = argin[0][:]
+        StorageLoggingDevice = argin[1][:]
+        i = 0
+        while i < len(StorageLoggingLevel[:]):
+            self.info_stream("Storage logging level : %s, Device : %s", StorageLoggingLevel[i], StorageLoggingDevice[i])
+            logger.debug("Storage logging level : %s, Device : %s", StorageLoggingLevel[i], StorageLoggingDevice[i])
+            logger.info("Storage logging level : %s, Device : %s", StorageLoggingLevel[i], StorageLoggingDevice[i])
+            logger.warning("Storage logging level : %s, Device : %s", StorageLoggingLevel[i], StorageLoggingDevice[i])
+            logger.error("Storage logging level : %s, Device : %s", StorageLoggingLevel[i], StorageLoggingDevice[i])
+            logger.fatal("Storage logging level : %s, Device : %s", StorageLoggingLevel[i], StorageLoggingDevice[i])
+            dev1 = DeviceProxy(StorageLoggingDevice[i])
```

Comment:
Change the variable name from dev1 to something more appropriate.

Reply:
Replaced it with devProxy.